src/Deserializers/SnakDeserializer.php

```diff
@@ -107,6 +107,8 @@ private function newSomeValueSnak( array $serialization ) {
 
 	private function newValueSnak( array $serialization ) {
 		$this->requireAttribute( $serialization, 'datavalue' );
+		// Do not deserialize derived value snaks
+		$this->assertNotAttribute( $serialization, 'derived-values' );
```

Comment:
This comment is somewhat misleading, as the line does not prevent deserialization of derived values. You can remove it, and such deserialization would still not happen. What it does is prevent deserializing something that has derived values in it altogether. What is the rationale behind doing that? If I'm not mistaken, we are simply ignoring such things elsewhere.